eitprocessing/eit_data/__init__.py

```diff
@@ -144,71 +148,75 @@ def _ensure_path_list(path: PathArg) -> list[Path]:
-        from .draeger import DraegerEITData  # pylint: disable=import-outside-toplevel
-        from .sentec import SentecEITData  # pylint: disable=import-outside-toplevel
-        from .timpel import TimpelEITData  # pylint: disable=import-outside-toplevel
+        from .draeger import DraegerEITData
+        from .sentec import SentecEITData
+        from .timpel import TimpelEITData
 
         vendor_classes: dict[Vendor, type[EITData_]] = {
             Vendor.DRAEGER: DraegerEITData,
             Vendor.TIMPEL: TimpelEITData,
             Vendor.SENTEC: SentecEITData,
         }
-        subclass = vendor_classes[vendor]
-        return subclass
+        return vendor_classes[vendor]
 
     @staticmethod
-    def _check_first_frame(first_frame):
+    def _check_first_frame(first_frame: int | None) -> int:
         if first_frame is None:
             first_frame = 0
         if int(first_frame) != first_frame:
-            raise TypeError(
-                f"`first_frame` must be an int, but was given as"
-                f" {first_frame} (type: {type(first_frame)})"
-            )
+            msg = f"`first_frame` must be an int, but was given as {first_frame} (type: {type(first_frame)})"
+            raise TypeError(msg)
         if first_frame < 0:
-            raise ValueError(
-                f"`first_frame` can not be negative, but was given as {first_frame}"
-            )
-        first_frame = int(first_frame)
-        return first_frame
+            msg = f"`first_frame` can not be negative, but was given as {first_frame}"
+            raise ValueError(msg)
+        return int(first_frame)
 
     @staticmethod
     def _ensure_vendor(vendor: Vendor | str) -> Vendor:
         """Check whether vendor exists, and assure it's a Vendor object."""
-        if not vendor:
-            raise NoVendorProvided()
-
         try:
             return Vendor(vendor)
         except ValueError as e:
-            raise UnknownVendor(f"Unknown vendor {vendor}.") from e
+            msg = f"Unknown vendor {vendor}."
+            raise UnknownVendorError(msg) from e
 
-    @classmethod
-    def concatenate(cls, a: T, b: T, label: str | None = None) -> T:
-        cls.isequivalent(a, b, raise_=True)
+    def concatenate(self: T, other: T, label: str | None = None) -> T:
+        cls = self.__class__
+        self.isequivalent(other, raise_=True)
 
-        a_path = cls._ensure_path_list(a.path)
-        b_path = cls._ensure_path_list(b.path)
+        a_path = cls._ensure_path_list(self.path)
+        b_path = cls._ensure_path_list(other.path)
         path = a_path + b_path
 
-        if np.min(b.time) <= np.max(a.time):
-            raise ValueError(f"{b} (b) starts before {a} (a) ends.")
-        time = np.concatenate((a.time, b.time))
+        if np.min(other.time) <= np.max(self.time):
+            msg = f"{other} (b) starts before {self} (a) ends."
+            raise ValueError(msg)
+        time = np.concatenate((self.time, other.time))
```

Comment:
I agree. I've been working the addition mixin. I just pushed 129c_addition_mixin_psomhorst* that solves this issue by adding a flag `check_time_consistency` that allows for overruling this check.

* I created a new branch instead of pushing to your branch, because it's rebased on this PR.